libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/route/RouteFetcher.java

```diff
@@ -71,31 +63,23 @@ public void updateRouteProfile(String routeProfile) {
    * @param routeProgress for remaining waypoints along the route
    * @since 0.13.0
    */
-  public void findRouteFromRouteProgress(Context context, Location location, RouteProgress routeProgress) {
-    if (isValidProgress(location, routeProgress)) {
+  public void findRouteFromRouteProgress(Location location, RouteProgress routeProgress) {
+    if (isInvalidProgress(location, routeProgress)) {
       return;
     }
     this.routeProgress = routeProgress;
-    NavigationRoute.Builder builder = buildRouteRequest(context, location, routeProgress);
+    NavigationRoute.Builder builder = buildRequestFromLocation(location, routeProgress);
     executeRouteCall(builder);
   }
 
-  private boolean isValidProgress(Location location, RouteProgress routeProgress) {
-    return location == null || routeProgress == null;
-  }
-
-  private NavigationRoute.Builder buildRouteRequest(Context context, Location location, RouteProgress routeProgress) {
+  @Nullable
+  private NavigationRoute.Builder buildRequestFromLocation(Location location, RouteProgress progress) {
+    if (contextWeakReference.get() == null) {
+      return null;
+    }
+    Context context = contextWeakReference.get();
```

Comment:
We can put this line at the beginning and avoid having to get the context here again 👀 
```java
Context context = contextWeakReference.get();
if (context == null) {
  return null;
}
// ...
NavigationRoute.Builder builder = NavigationRoute.builder(context)
```